### PR TITLE
Upgrade kube-dns to 1.14.1

### PR DESCRIPTION
**Pull request by r2d4**

### CODE REVIEW

File: deploy/addons/kube-dns/kube-dns-cm.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: kube-dns
+  namespace: kube-system
+  labels:
+    addonmanager.kubernetes.io/mode: EnsureExists
```
Comment (aaron-prindle) [inline deploy/addons/kube-dns/kube-dns-cm.yaml]: nit: newline

File: deploy/addons/kube-dns/kube-dns-controller.yaml [hunk 2]
```diff
@@ -68,59 +74,84 @@ spec:
...
-        - containerPort: 8080
+        - containerPort: 10054
+          name: metrics
           protocol: TCP
-      dnsPolicy: Default  # Don't use cluster DNS.
+        resources:
+          requests:
+            memory: 20Mi
+            cpu: 10m
+      dnsPolicy: Default  # Don't use cluster DNS.
```
Comment (aaron-prindle) [inline deploy/addons/kube-dns/kube-dns-controller.yaml]: nit: newline

File: deploy/addons/kube-dns/kube-dns-cm.yaml (hunk 1 above)
Comment (dlorenc) [inline deploy/addons/kube-dns/kube-dns-cm.yaml]: Any reason to use this instead of Reconcile?

Comment (r2d4) [inline deploy/addons/kube-dns/kube-dns-cm.yaml]: This is kind of an placeholder configmap.   I think you're supposed to update it and somehow restart DNS to pick up those changes?  Taken from upstream and there weren't any additional justifications https://github.com/kubernetes/kubernetes/pull/40108

Comment (r2d4) [inline deploy/addons/kube-dns/kube-dns-cm.yaml]: fixed

File: deploy/addons/kube-dns/kube-dns-controller.yaml (hunk 2 above)
Comment (r2d4) [inline deploy/addons/kube-dns/kube-dns-controller.yaml]: fixed

**Comment by r2d4** [comment]
@minikube-bot retest this please

**Pull request merged**